src/index.js

```diff
@@ -50,4 +30,95 @@ const renderList = (list) => {
   ToDoListContainer.innerHTML = innerList;
 };
 
-window.onload = renderList(todoList);
\ No newline at end of file
+const saveToLocalStorage = (list) => {
+  localStorage.setItem('todoList', JSON.stringify(list));
+};
+
+const addTask = (description) => {
+  const newTask = {
+    id: Date.now(),
+    description,
+    completed: false,
+    index: todoList.length + 1,
+  };
+  todoList.push(newTask);
+  saveToLocalStorage(todoList);
+  renderList(todoList);
+};
+
+const editTaskDescription = (taskId, newDescription) => {
+  todoList.forEach((task) => {
+    if (task.id === taskId) {
+      task.description = newDescription;
+    }
+  });
+  saveToLocalStorage(todoList);
+  renderList(todoList);
+};
```

Comment:
I can see that you have implemented the edit functionality here, however, it is not working as expected, consider making the item editable when it is clicked upon in order to make it possible for a user to perform edits